lightpack.py

```diff
@@ -173,13 +184,123 @@ def _sendAndExpectSuccess(self, command):
 		"""
 		self._sendAndExpect(command, '%s:success' % self._name(command))
 
-	def getProfiles(self):
+	def getColour(self, led):
+		"""
+		Get the specified LED's colour.
+
+		:param led: 0-based LED index or its preconfigured alias
+		:type led: str or int
+		:returns: Tuple of red, green, blue values (0 to 255)
+		"""
+		return self.getColoursFromAll()[self._ledIndex(led)]
+	getColor = getColour
+
+	def getColours(self, *args):
+		"""
+		Get the individual colours of multiple LEDs.
+
+		:returns: Dictionary of tuples of red, green, blue values (0 to 
+		255), using LED numbers as integer keys
+		"""
+		defs = [self._ledIndex(arg) for arg in args]
+		colours = self.getColoursFromAll()
+		return dict([(k, colours[k]) for k in colours if k in defs])
+	getColors = getColours
+
+	def _ledColourRead(self, snippet):
+		"""
+		Read a LED colour state snippet into RGB Tuple.
+
+		:param snippet: LED colour state snipped
+		:type snippet: str
+		:returns: Tuple with LED and tuple of red, green, blue values (0 to 
+		255)
+		"""
+		try:
+			parts = snippet.split('-', 1)
+			colours = parts[1].split(',', 2)
+			rgb = [int(x) for x in colours if x.strip()]
+			led = parts[0].strip()
+			if not led or len(rgb) != 3:
+				return None
+			return int(led), tuple(rgb)
+		except IndexError:
+			return None
+		except ValueError:
+			return None
```

Comment:
Is silently failing the right thing to do on these error conditions? (The same is done later in `getScreenSize`, and other methods.)

Reply:
If memory serves me well, the API was randomly belching erroneous snippets sometimes like `1-255,,128`. The silent exception handling allowed to simplify the code instead of many if/else checks, but should be raise a proper notice instead? Also, maybe the ValueError isn't needed here anymore since L224 is doing a sanity check.

Reply:
Hm if the API really is that noisy maybe silently ignoring is the right thing to do. Can you go ahead and double check whether it's necessary to ignore ValueErrors? Whatever errors are ignored in the end, it's probably worth documenting (with a comment perhaps) so we know why things are being silently ignored.

Reply:
Well, I am not able to make the API choke on its snippet responses anymore when stress-testing it, even in a busy network pipe. I can read the odd response when reverting to a non-buffered connection, but nothing pertaining to colour or size snippets.

I've therefore simplified this method and its calling one, seeing as there is no real need to shield against non-reproducible (did I dream? 😯) errors.

The same treatment was applied to the similarly introduced `_ledDimensionRead()` method, which I've also renamed to `_ledSizeRead()` for uniformity's sake.